admin-dev/themes/new-theme/js/pages/order/create/cart-rule-manager.js

```diff
@@ -90,6 +90,7 @@ export default class CartRuleManager {
     EventEmitter.on(eventMap.cartRuleAdded, (cartInfo) => {
       const cartIsEmpty = cartInfo.products.length === 0;
       this.cartRulesRenderer.renderCartRulesBlock(cartInfo.cartRules, cartIsEmpty);
+      this.productRenderer.renderList(cartInfo.products);
```

Comment:
Why do you need to update the product list here? Does the cart rule affect the product rendering?

Reply:
Oh I get it, it's cart rule that adds the gift product right? Hence the products re-rendering

Reply:
yes. And yes this is wip. I only rebased it now. Its wip because of the last found bug.